language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionUtil.java

```diff
@@ -187,17 +189,40 @@ public static List<String> getPossibleTypes(TypeSymbol typeDescriptor, List<Text
      */
     public static List<String> getPossibleTypes(TypeSymbol typeDescriptor, CodeActionContext context,
                                                 ImportsAcceptor importsAcceptor) {
+        return new ArrayList<>(getPossibleTypeSymbols(typeDescriptor, context, importsAcceptor).values());
+    }
+
+    public static Map<TypeSymbol, String> getPossibleTypeSymbols(TypeSymbol typeDescriptor,
+                                                                 List<TextEdit> importEdits,
+                                                                 CodeActionContext context) {
+        ImportsAcceptor importsAcceptor = new ImportsAcceptor(context);
+        Map<TypeSymbol, String> possibleTypes = getPossibleTypeSymbols(typeDescriptor, context, importsAcceptor);
+        importEdits.addAll(importsAcceptor.getNewImportTextEdits());
+        return possibleTypes;
+    }
+
+    public static Map<TypeSymbol, String> getPossibleTypeSymbols(TypeSymbol typeDescriptor, CodeActionContext context,
+                                                                 ImportsAcceptor importsAcceptor) {
+        Optional<SemanticModel> semanticModel = context.currentSemanticModel();
+        if (semanticModel.isEmpty()) {
+            return Collections.emptyMap();
+        }
         if (typeDescriptor.getName().isPresent() && typeDescriptor.getName().get().startsWith("$")) {
             typeDescriptor = CommonUtil.getRawType(typeDescriptor);
         }
 
-        List<String> types = new ArrayList<>();
+        Map<TypeSymbol, String> typesMap = new HashMap<>();
+        Types types = semanticModel.get().types();
```

Comment:
We should use a new `TypeBuilder` instance per new type being created. So, we cannot reuse it.

Reply:
I have used this approach in the `buildUnionOfIterables()` method in the `ContextTypeResolver`.  Why do we need to create new instance every time? As I remember after creating type symbols it will go back to the start state.

Reply:
@sanjana

Reply:
@IMS94, since the caching mechanism was removed within the builders, we could use the same builder instance as suggested by @LakshanWeerasinghe.